src/main/java/com/redhat/exhort/providers/GradleProvider.java

```diff
@@ -0,0 +1,205 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.exhort.providers;
+
+import com.redhat.exhort.Api;
+import com.redhat.exhort.Provider;
+import com.redhat.exhort.logging.LoggersFactory;
+import com.redhat.exhort.sbom.Sbom;
+import com.redhat.exhort.sbom.SbomFactory;
+import com.redhat.exhort.tools.Ecosystem.Type;
+import com.redhat.exhort.tools.Operations;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.redhat.exhort.impl.ExhortApi.debugLoggingIsNeeded;
+
+/**
+ * Concrete implementation of the {@link Provider} used for converting dependency trees
+ * for Gradle projects (gradle.build) into a content Dot Graphs for Stack analysis or Json for
+ * Component analysis.
+ **/
+public final class GradleProvider extends BaseJavaProvider {
+
+  private Logger log = LoggersFactory.getLogger(this.getClass().getName());
+  public GradleProvider() {
+    super(Type.GRADLE);
+  }
+
+  @Override
+  public Content provideStack(final Path manifestPath) throws IOException {
+    Path tempFile = getDependencies(manifestPath);
+    if (debugLoggingIsNeeded()) {
+      String stackAnalysisDependencyTree = Files.readString(tempFile);
+      log.info(String.format("Package Manager Maven Stack Analysis Dependency Tree Output: %s %s", System.lineSeparator(), stackAnalysisDependencyTree));
+    }
+    Map<String, String> propertiesMap = extractProperties(manifestPath);
+
+    var sbom = buildSbomFromTextFormat(tempFile, propertiesMap, "runtimeClasspath");
+    return new Content(sbom.getAsJsonString().getBytes(), Api.CYCLONEDX_MEDIA_TYPE);
+  }
+
+  private static Path getDependencies(Path manifestPath) throws IOException {
+    // check for custom gradle executable
+    var gradle = Operations.getCustomPathOrElse("gradle");
+    // create a temp file for storing the dependency tree in
+    var tempFile = Files.createTempFile("exhort_graph_", null);
+    // the command will create the dependency tree in the temp file
+    String gradleCommand = gradle + " dependencies";
+
+    String[] cmdList = gradleCommand.split("\\s+");
+    String gradleOutput = Operations.runProcessGetOutput(Path.of(manifestPath.getParent().toString()), cmdList, null);
+    Files.writeString(tempFile, gradleOutput);
+
+    return tempFile;
+  }
+
+  private Path getProperties(Path manifestPath) throws IOException {
+    Path propsTempFile = Files.createTempFile("propsfile", ".txt");
+    var gradle = Operations.getCustomPathOrElse("gradle");
+    String propCmd = gradle + " properties";
+    String[] propCmdList = propCmd.split("\\s+");
+    String properties = Operations.runProcessGetOutput(Path.of(manifestPath.getParent().toString()), propCmdList, null);
```

Comment:
No need to call method runProcessGetOutput/3 with last arguments as null, as this is exactly what method runProcessGetOutput/2 does. 
```suggestion
String properties = Operations.runProcessGetOutput(Path.of(manifestPath.getParent().toString()), propCmdList);
```

Reply:
Done